### ISSUE TITLE
support for NOT NULL constraint when using management=True and the Spatialite extension to SQLite

### ISSUE THREAD

**Issue by StephenWithPH** [geoalchemy/geoalchemy2#326]
Spatialite's [SQL function reference](https://www.gaia-gis.it/gaia-sins/spatialite-sql-latest.html#p16) for the `AddGeometryColumn` function defines its signature as:

```sql
AddGeometryColumn(
  table String
  , column String
  , srid Integer
  , geom_type String
  [ , dimension String [ , not_null Integer ] ] 
) : Integer
```

The docs explain `[ , not_null Integer ]`:

```
the optional 6th arg [not_null] is a non-standard extension required by the peculiar SQLite arch:

    if set to 0 [false], then the Geometry column will accept NULL values as well. This is the default behaviour
    if set to any <> 0 value [true], then the Geometry will be defined using a NOT NULL clause
```

[class geoalchemy2.types.Geometry](https://geoalchemy-2.readthedocs.io/en/latest/types.html#geoalchemy2.types.Geometry) and its base type [class geoalchemy2.types._GISType](https://geoalchemy-2.readthedocs.io/en/latest/types.html#geoalchemy2.types._GISType) do not support a `not_null` argument.

Falling back on standard SQLAlchemy syntax in a model...

```py
Base: Any = declarative_base()

class Thing(Base):
    __tablename__ = "thing"
    thing_id = Column(Integer, name="id", primary_key=True)
    geometry = Column(
        Geometry(geometry_type="MULTIPOLYGON", management=True, srid=4326),
        nullable=False,
    )

```

... does not get the job done. The generated SQL is:

```
INFO  [sqlalchemy.engine.Engine] 
CREATE TABLE thing (
        id INTEGER NOT NULL, 
        PRIMARY KEY (id)
)

INFO  [sqlalchemy.engine.Engine] SELECT AddGeometryColumn(?, ?, ?, ?, ?) AS "AddGeometryColumn_1"
INFO  [sqlalchemy.engine.Engine] [no key 0.00017s] ('thing', 'geometry', 4326, 'MULTIPOLYGON', 2)     <---- no sixth parameter
```

Would it be possible to get GeoAlchemy2 to support this use case?

**Comment by adrien-berchet** [comment]
Hello @StephenWithPH 
I tried something in #327 , does it fit your need?

**Comment by adrien-berchet** [comment]
And thanks for you very clear report! :)

**Comment by StephenWithPH** [comment]
@adrien-berchet ... works like a charm! Thank you for the PR!

`pyproject.toml`:
```toml
[tool.poetry.dependencies]
GeoAlchemy2 = { git = "https://github.com/adrien-berchet/geoalchemy2.git", rev = "1965a983a835e20156a918a649111d933f5357fb" }
```

`model.py`:
```py
Base: Any = declarative_base()

class Thing(Base):
    __tablename__ = "thing"
    thing_id = Column(Integer, name="id", primary_key=True)
    geometry = Column(
        Geometry(geometry_type="MULTIPOLYGON", management=True, srid=4326, nullable=False),
    )
```

logs:
```text
INFO  [sqlalchemy.engine.Engine] 
CREATE TABLE thing (
        id INTEGER NOT NULL, 
        PRIMARY KEY (id)
)

INFO  [sqlalchemy.engine.Engine] [no key 0.00009s] ()
INFO  [sqlalchemy.engine.Engine] SELECT AddGeometryColumn(?, ?, ?, ?, ?, ?) AS "AddGeometryColumn_1"
INFO  [sqlalchemy.engine.Engine] [no key 0.00019s] ('fire', 'geometry', 4326, 'MULTIPOLYGON', 2, 1)    <---- that last '1' is what we needed!
```

**Comment by adrien-berchet** [comment]
Perfect!
I merged #327 so you can use the master branch for now. I still have to look at #328 before releasing a new version, I will try to do it next week.

**Comment by adrien-berchet** [comment]
The version 0.9.3 with this feature is now released.
